ui/lib/replication/addon/controllers/application.js

```diff
@@ -105,7 +116,26 @@ export default Controller.extend(copy(DEFAULTS, true), {
     onSubmit(/*action, mode, data, event*/) {
       return this.submitHandler(...arguments);
     },
+    copyClose(successMessage) {
+      // separate action for copy & close button so it does not try and use execCommand to copy token to clipboard
+      if (!!successMessage && typeof successMessage === 'string') {
+        this.get('flashMessages').success(successMessage);
+      }
+      this.toggleProperty('isModalActive');
+      this.transitionToRoute('mode.secondaries');
+    },
+    toggleModal(successMessage) {
+      if (!!successMessage && typeof successMessage === 'string') {
+        this.get('flashMessages').success(successMessage);
+      }
+      // use copy browser extension to copy token if you close the modal by clicking outside of it.
+      const htmlSelectedToken = document.querySelector('.textarea');
```

Comment:
You may consider using the `<CopyButton>` component on the modal and then hooking into closing the modal close on the success action, the querySelector seems a little tenuous if there's ever more than one element with `.textArea` (see app/templates/components/secret-edit.hbs for an example)

Reply:
yes -- to echo chelsea's advice i'd recommend using the `HoverCopyButton` component & using the same codeblock styles that we use when generating an operation token and elsewhere in the UI. the codeblock (vs an input) makes it clearer that this value is meant to be copied and not edited. here's an example of how to do that:
https://github.com/hashicorp/vault/blob/061bc8ed2bbed49eb4f54c3a04647fbbefb13e48/ui/app/templates/components/shamir-flow.hbs#L23-L34

![image](https://user-images.githubusercontent.com/903288/83816826-2aa58b80-a678-11ea-8e37-e554d1279538.png)

Reply:
I chatted this through with Chelsea.  I originally tried this solution, however, this modal is a bit unique from the others.  After chatting with design, we decided that we wanted the modal to copy the token if you click *off* the modal (not just on a button click).  The reason is that the moment you click "Generate Token" (so before the modal even opens) a token has been generated.  If you click off the modal and we don't copy the token or redirect them to the secondaries page, then if they try and click "Generate Token" again they'll get an error that the token already exists.  Thus, we really want to force them, no matter what they do, to copy the token and redirect to the secondaries page.  Additionally, because the `@onClose` action of the modal needs to both copy and send the success message, I moved this to an action inside the controller.  And the `@clipboardText` is only an hbs helper, and is not something I can use in the controller.  Hence the whole action -> select and copy the token + send a success message flow.  

Design originally had the copy button icon as you both mentioned, but again, because we needed them to copy the token when they closed the modal, we removed it as it was redundant.  Right now, they have two ways to copy the token, 1. click the copy & close button, or 2. click off the modal (e.g. close it).  We didn't need another copy button that ultimately no matter what their next step was, would copy the token again.  

Let me know if that makes sense.